packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -240,9 +240,7 @@ export class PreferencesController extends BaseController<
             accounts.add(account);
           }
         }
-        if (accounts.size > 0) {
-          this.syncIdentities(Array.from(accounts));
-        }
+        this.#syncIdentities(Array.from(accounts));
```

Comment:
We are changing a behavior here, as identity sync won't be skipped when locking the wallet anymore.

Currently, an error is thrown when calling `syncIdentities` with an empty array. As with these changes `#syncIdentities` becomes callable only internally, throwing an error in the function may make less sense (because maybe eventually we will want to empty out identities intentionally), but perhaps we should still skip it in this event handler, and allow `#syncIdentities` to empty out the identities array (i.e. accepting an empty array)

That being said, I'm not sure if this change should be done in a separate PR and be tracked by a separate issue, as we'll eventually want to do the same on the extension's PreferencesController

Reply:
I just updated `#syncIdentities` to allow the process in case of an empty array to empty identities (move them to `lostIdentities`) when called with an empty array